### ISSUE TITLE
[DOC] The CLI parameter list is incomplete

### ISSUE THREAD

**Issue by attachmentgenie** [grafana/tempo#1791]
**Is your feature request related to a problem? Please describe.**
I am trying setup a new Tempo system that will run the following config (excerpt)

```
multitenancy_enabled: false
server:
  http_listen_port: 3200
memberlist:
  join_members:
    - dnssrv+_tempo-gossip._tcp.service.consul
distributor:
  receivers:
    zipkin:
    otlp:
      protocols:
        http:
        grpc:
storage:
  trace:
    backend: s3
    s3:
      bucket: traces
      endpoint: s3.service.consul
      access_key: nottherealkey
      secret_key: nottherealsecret
      insecure: true
```

but if you try and port this setup to only using CLI parameters you will find that several parameters are current not implemented in the Tempo CLI, more specifically the following 2 are absent that are available in e.g the Mimir CLI, The distributor section seems absent entirely

```
    -memberlist.join=dnssrv+_mimir-gossip._tcp.service.consul
    -storage.trace.s3.insecure=true
```

**Describe the solution you'd like**
Support the required cli parameters, similar to how mimir does it

**Describe alternatives you've considered**
keep using config file

**Additional context**

**Comment by joe-elliott** [comment]
So the original idea was to only expose the most used configuration options using the CLI. We were trying to simplify the options printed when the user dumped help to make Tempo easier to understand. As an operator "graduated" from simple to complex deployments we expected them to use yaml for configuration.

Whether or not these goals have been achieved is difficult to say. We would be open to making all configuration options available on the CLI if the community wants it.

**Comment by attachmentgenie** [comment]
Right, than my learning/maturity path is just weird.  I generally switch the other way when i need/want to run in a non trivial way, that way i dont have to drag around a config file with my dockerfile/nomad files.

**Comment by joe-elliott** [comment]
> Right, than my learning/maturity path is just weird

I wouldn't necessarily say that. The initial choice to not include every CLI parameter was a gut call and not backed by much in the way of data. This is really some of the first feedback we've received on it.

As Tempo has grown I am tempted to add all CLI params.

**Comment by joe-elliott** [comment]
Marking this `keepalive`. Looking to continue discussing this if other community members have strong feelings.

**Comment by EraYaN** [comment]
tempo-cli should mirror other grafana labs CLI projects in as may ways as possible, so if all those projects have all the parameters so should tempo-cli. And this should hold for everything, in how to install, use, parse output etc.

**Comment by knylander-grafana** [comment]
Depending on the decision in the community, we can update the docs to better cover CLI usage for how to install, use, parse, etc.

**Comment by knylander-grafana** [comment]
We updated the docs to make sure we had the options completed. These updates were done in this PR: https://github.com/grafana/tempo/pull/6345/
